### ISSUE TITLE
Unnecessary doubled souce package

### ISSUE THREAD

**Issue by tansy** [dbry/WavPack#176]
Found that sources fot windows and linux are separate and after closer inspection I can't shake the impression that it's unnecessary hastle. You have to generate two differenct packages which do not collide in any way anyway. One source package that includes everything is more than enough. One can generate relevant building scripts exacly same way as if they were separate.

Also maintaining DOS line endings (CRLF) is unnecessary either. Also irritating when seen or edited under *nix. Diff for that matter threats two same lines, but with different line endins, as different. Even [microsoft doesn't use DOS endings](https://github.com/microsoft/Windows-classic-samples/raw/main/Samples/AmbientLightAware/cpp/AmbientLightAware.cpp) any more. And for many years now.
It could be solved with simple \`.gitattributes'. I can do that (i mean PR) if you accept that.

**Comment by dbry** [comment]
Thanks for the message!

> Found that sources fot windows and linux are separate and after closer inspection I can't shake the impression that it's unnecessary hastle. You have to generate two differenct packages which do not collide in any way anyway. One source package that includes everything is more than enough. One can generate relevant building scripts exacly same way as if they were separate.

The idea is that the *nix tarball should take case of everything now. I used to have a separate source package for Windows, but that's going away for the next release.
 
> 
> Also maintaining DOS line endings (CRLF) is unnecessary either. Also irritating when seen or edited under *nix. Diff for that matter threats two same lines, but with different line endins, as different. Even [microsoft doesn't use DOS endings](https://github.com/microsoft/Windows-classic-samples/raw/main/Samples/AmbientLightAware/cpp/AmbientLightAware.cpp) any more. And for many years now. It could be solved with simple `.gitattributes'. I can do that (i mean PR) if you accept that.

Hmm. I have certainly not heard that Microsoft is ditching CRLF, but Visual Studio since at least 2019 certainly accepts both without an issue (although it will create _new_ files with CRLF). In the WavPack sources any file edited within the last decade should be *nix line endings unless I missed one. There are certainly legacy files in there (some I did not even create) that might still have DOS endings, but is that really an issue? In any event, I would be happy to look at a .gitattribute file that "fixes" this.

**Comment by tansy** [comment]
> Hmm. I have certainly not heard that Microsoft is ditching CRLF

Depends what you mean by 'ditch'. They still support it but not force it and not use it in their published sources, which were linked.

They don't not force it. If you use MSDEV from VC6 and read LF source, it will compile, problem, and will not be convertred into CRLF either. I was testing it and that's the result I got: [zipped project](https://files.fm/f/a62jeuurgq), [screenshot](https://i.postimg.cc/cJfc4R9Z/Hi-h-dsp.png).


>  I would be happy to look at a .gitattribute file that "fixes" this.

I get the irony. It obviously will not 'fix' what is already in a repo, it would only 'fix' whatever is uploaded since.

file could look like this:
```
# Set the default behavior
* text=auto

# Explicitly declare source files
*.c text eol=lf
*.cc text eol=lf
*.h text eol=lf
*.sh text eol=lf

# Denote files that should not be modified
*.odt binary
*.pdf binary
*.png binary
*.wav binary
*.wv  binary
*.zip binary

# Visual Studio
*.dsp text eol=crlf
*.dsw text eol=crlf
*.sln text eol=lf
*.vcxproj* text eol=lf
*.vcproj* text eol=lf
*.suo binary
*.res binary
*.rc text eol=lf
```

or even `* text=auto eol=lf` and list binaries only.

When it comes to files in repository I would use the [script](https://gist.githubusercontent.com/tansy/c4a1f7ea1019915b4263c024757d3ac4/raw) that was designed to solve that problem before.
The result would look like [this](https://files.fm/f/635uyqakvt).

> but is that really an issue?

If you decide to use .gitattributes it would be logical to 'normalize' the files before, or they will be somewhat mixed on the way, when any modifications occur.

**Comment by dbry** [comment]
I checked the repos and pretty much verified what I had guessed. The only files still using DOS endings are legacy files that I didn’t even create and that are really only present for historical reference (e.g., `winamp/in.h`) and also files generated by MSVC (e.g., `wavpackdll/wavpackdll.rc`). When I converted those to unix line endings and edited them with `Visual Studio 2019` they ended up with mixed line endings, which I think is a little weird and undesirable.

I probably don’t fully understand this problem, and I appreciate your input, but at this point I think I’ll leave everything as is but keep the issue open as a reminder. Thanks!

[Also some interesting/funny perspectives here.](https://github.com/microsoft/Windows-Dev-Performance/issues/85)

**Comment by tansy** [comment]
Ok, ok, i get that. Maybe, at least .gitattributes, so if you _edited them with Visual Studio 2019 they ended up with mixed line endings_?

----
> [Also some interesting, funny predictive](https://github.com/microsoft/Windows-Dev-Performance/issues/85)

It really is funny, because they use while bunch of slogans, to maintain status quo.
Like this one:
>  this behavior is hard-coded into hundreds of millions of apps, tools, scripts, etc. many of which will break if CRLF is replaced with LF, again, 

Yeah, right. Even .bat files don't need to use CRLF, nor to mention anything else.

And that:

> likely crashing millions of systems, businesses, and even several nations.

Made me rotfl.

**Comment by dbry** [comment]
> And that:
> 
> > likely crashing millions of systems, businesses, and even several nations.
> 
> Made me rotfl.

Exactly! I assume he means that it would only crash fairly small nations...  :smile_cat:
